frcxx_nft/src/lib.rs

```diff
@@ -227,8 +321,14 @@ where
             Err(_) => return Ok(()), // if operator didn't exist this is a no-op
         };
 
-        self.state.revoke_for_all(&self.bs, owner, operator)?;
-        Ok(())
+        self.transaction(|state, bs| {
+            let mut owner_map = state.get_owner_data_hamt(bs)?;
+
+            state.revoke_for_all(&mut owner_map, owner, operator)?;
+
+            state.owner_data = owner_map.flush().map_err(StateError::from)?;
```

Comment:
This kind of thing clearly has no business logic so I think the old way was fine.